src/bomf/model/__init__.py

```diff
@@ -0,0 +1,61 @@
+"""
+general data models for migrations
+"""
+from typing import Any, Iterable, Optional, Protocol, Type, TypeVar
+
+import attrs
+import bo4e.bo.geschaeftsobjekt
+
+
+# pylint:disable=too-few-public-methods
+@attrs.define(kw_only=True, auto_attribs=True)
+class BusinessObjectRelation:
+    """
+    A business object relation describes the relation between two business object.
+    E.g. a relation could have the type "has_melo" where relation_part_a is a bo4e.bo.Vertrag
+    and relation_part_b is a bo4e.bo.Messlokation. Some relations are already defined in BO4E itself (e.g MaLo/MeLo)
+    or MeLo/Address.
+    The idea is to not enforce too much of a structure to the downstream code but still push coders to think about
+    necessary relation information.
+    """
+
+    relation_type: str = attrs.field(validator=attrs.validators.instance_of(str))
```

Comment:
Finde es noch schwierig wie hier die relation festgehalten werden soll.
Einfach in Prosa oder gibt es eine gewisse Systax die man einhalten kann/soll?